torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1947,9 +1944,12 @@ Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const T
 
   if (gu.defined()) {
     auto guh = gu.conj().transpose(-2, -1);
-    u_term = at::matmul(u, at::matmul(F.mul(at::matmul(uh, gu) - at::matmul(guh, u)), sigma_mat));
+    u_term = at::matmul(u, (F.mul(at::matmul(uh, gu) - at::matmul(guh, u)) * sigma.unsqueeze(-2)));
     if (m > k) {
-      u_term = u_term + at::matmul(im - at::matmul(u, uh), at::matmul(gu, sigma_mat_inv));
+      // projection operator onto subspace orthogonal to span(U) defined as I - UU^H
+      auto proj_on_ortho_u = -at::matmul(u, uh);
+      proj_on_ortho_u.diagonal(/*offset=*/0, /*dim1=*/-2, /*dim2=*/-1).add_(1);
```

Comment:
Is this function supposed to be double differentiable? This inplace add might cause issues if we try to autodiff through this.

Reply:
Hmm, good question, but I did the same trick for lobpcg backwards and it did not cause any issues...

Reply:
If gradgradcheck is happy, I'm happy :D